services/system/Accounts/ui/src/components/AccountSelection.tsx

```diff
@@ -0,0 +1,722 @@
+          </DialogContent>
+          <div className="max-w-lg mx-auto">
+            <div className="text-center text-muted-foreground py-2">
+              <span>
+                {isInitialLoading
+                  ? "Loading..."
+                  : isNoAccounts
+                  ? `No accounts available.`
+                  : isInvite
+                  ? `Select an account to accept invite to `
+                  : `Select an account to login to `}
+                {!isNoAccounts && (
+                  <span className="text-primary">{appName}</span>
+                )}
+              </span>
+            </div>
+            {!isNoAccounts && (
+              <div className="relative ml-auto flex-1 md:grow-0 mb-3">
+                <Search className="absolute left-2.5 top-2.5 h-4 w-4 text-muted-foreground" />
+                <Input
+                  value={activeSearch}
+                  type="search"
+                  onChange={(e) => setActiveSearch(e.target.value)}
+                  placeholder="Search..."
+                  className="w-full rounded-lg bg-background pl-8 "
+                />
+              </div>
+            )}
+
+            <div className="flex flex-col gap-3 ">
+              <div className="flex flex-col gap-3 max-h-[600px] overflow-auto">
+                {isAccountsLoading
+                  ? [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14].map(
+                      (_, index) => (
+                        <Account
+                          onSelected={() => {}}
+                          isSelected={false}
+                          key={index}
+                        />
+                      )
+                    )
+                  : accountsToRender.map((account) => (
+                      <Account
+                        onSelected={() => {
+                          setSelectedAccountId(account.id);
+                        }}
+                        isSelected={selectedAccountId == account.id}
+                        key={account.id}
+                        name={account.account}
+                      />
+                    ))}
+              </div>
+              {isInvite && !isInviteClaimed && (
+                <button
+                  onClick={() => {
+                    setIsCreatingAccount(true);
+                    setIsModalOpen(true);
+                  }}
+                  className="flex p-4 shrink-0 items-center justify-center rounded-md border border-neutral-600 text-muted-foreground hover:text-primary hover:underline border-dashed"
+                >
+                  Create a new account
+                </button>
+              )}
+            </div>
+            {!isNoAccounts && (
+              <div className="my-3">
+                <Button
+                  disabled={!selectedAccount || isTxInProgress}
+                  onClick={() => {
+                    if (isInvite) {
+                      acceptInvite(z.string().parse(token));
```

Comment:
The invite is actually accepted as soon as this modal is dealt with:
![image](https://github.com/user-attachments/assets/fa8ef45e-b434-4c8d-9d35-b454084d7463)

Which means that going back to this screen afterwards is incorrect:
![image](https://github.com/user-attachments/assets/fd848e3d-8512-4d9a-aae6-1a5ba5e3074d)

There is actually no further opportunity to change your selection. If you created a new account, then you accepted the invite with that account

Reply:
For example. Create an account (from an invite), and then close the tab before hitting the subsequent Accept/Reject button, and then reload invite link. 

It will (correctly) tell you that the invite was already accepted with the new account.

Reply:
Will immediately go to next page now.